pkg/aws/aws.go

```diff
@@ -227,6 +227,15 @@ func (a *AWS) Collect(ch chan<- prometheus.Metric) {
 	providerScrapesTotalCounter.WithLabelValues(subsystem).Inc()
 }
 
+func (a *AWS) CheckReadiness() bool {
+	for _, c := range a.collectors {
+		if !c.CheckReadiness() {
+			return false
+		}
```

Comment:
I wonder if this is better suited to run in parallel, especially given some may take quite some time. If one of them returns false, shortcircuit and return false. Is that something that's possible?

Reply:
I think we should probably keep it serial for now, if we implement more complicated readiness checks that checking a lock, this may be worth the overhead of a goroutine